spec/models/authenticator/ldap_spec.rb

```diff
@@ -163,6 +164,19 @@ def authenticate
     let(:username) { 'alice' }
     let(:password) { 'secret' }
 
+    context "when using LDAP" do
+      let(:config) { super().merge(:ldap_role => true) }
+
+      it "encrypts password for queuing" do
+        called_encrypt_ldap_password = false
+        allow(MiqTask).to receive(:create).and_return(double(:id => :return_value))
+        allow(MiqQueue).to receive(:put) { :return_value }
+        allow(subject).to receive(:encrypt_ldap_password) { called_encrypt_ldap_password = true }
+        authenticate
+        expect(called_encrypt_ldap_password).to be_truthy
```

Comment:
Indeed, line 174 means this test always passes.

You should be doing some form of `expect(subject).to receive(:encrypt_ldap_password)`

I'm also not sure what `double(:id => :return_value)` does. Does RSpec have a `:return_value`? Even if it does, I'm not sure what you're trying to do in 172 or 173 (the latter would theoretically 'allow' MiqQueue something that it already does normally ;) )